### PR TITLE
AP_Mount: all backends send gimbal-device-attitude-status

### PR DESCRIPTION
**Pull request by rmackay9**
This updates all backends to send the [GIMBAL_DEVICE_ATTITUDE_STATUS](https://github.com/ArduPilot/mavlink/blob/master/message_definitions/v1.0/common.xml#L5637) message instead of the older ardupilot specific [MOUNT_STATUS](https://github.com/ArduPilot/mavlink/blob/master/message_definitions/v1.0/ardupilotmega.xml#L1218) message.

Some other changes are also included:

- autotests adjusted to consume the GIMBAL_DEVICE_ATTITUDE_STATUS message (instead of MOUNT_STATUS)
- GCS_MAVLink bug fix to forward MAV_CMD_DO_GIMBAL_MANAGER_PITCHYAW to the AP_Mount library.  Without this the gimbal responded to mission commands correctly but not to the equivalent immediate commands sent from a GCS in real-time.
- autotest updated to test the above command.  Note this required slightly relaxing one tests acceptance threshold probably because of the use of Quaternions instead of Eulers
- autotest gets a new set-mount-mode method to reduce duplicate code
- copter-gimbal.parm parameter file added to ease setting up a gimbal in SITL

This has been tested in SITL for servo and Gremsy backends to ensure they send the correct messages.  We currently have no way to test the other gimbal drivers (Solo, SToRM32 serial and mavlink, AlexMos serial) but they should be fine.
image

### CODE REVIEW

File: libraries/AP_Mount/AP_Mount_Alexmos.cpp [hunk 1]
```diff
@@ -102,15 +103,31 @@ bool AP_Mount_Alexmos::has_pan_control() const
...
+    q.from_euler(radians(_current_angle.x), radians(_current_angle.y), radians(_current_angle.z));
+    const float quat_array[4] = {q.q1, q.q2, q.q3, q.q4};
+
+    mavlink_msg_gimbal_device_attitude_status_send(chan,
+                                                   0,   // target system
+                                                   0,   // target component
+                                                   AP_HAL::millis(),    // autopilot system time
+                                                   get_gimbal_device_flags(),
+                                                   quat_array,      // attitude expressed as quaternion
+                                                   AP::logger().quiet_nanf(),    // roll axis angular velocity (NaN for unknown)
```
Comment (peterbarker) [inline libraries/AP_Mount/AP_Mount_Alexmos.cpp:127]: We use `std::numeric_limits<double>::quiet_NaN()` elsewhere when constructing mavlink packets.  Pretty sure we don't want to be using something from `AP_Logger` when constructing mavlink packets.

Comment (rmackay9) [reply libraries/AP_Mount/AP_Mount_Alexmos.cpp]: Txs for the review,fixed!

File: libraries/AP_Mount/AP_Mount_SoloGimbal.cpp [hunk 2]
```diff
@@ -115,15 +115,31 @@ void AP_Mount_SoloGimbal::set_mode(enum MAV_MOUNT_MODE mode)
...
-    }
-
     // block heartbeat from transmitting to the GCS
     GCS_MAVLINK::disable_channel_routing(chan);
+
+    if (!_gimbal.aligned()) {
+        return;
+    }
+
+    // construct quaternion from angle targets
```
Comment (tridge) [inline libraries/AP_Mount/AP_Mount_SoloGimbal.cpp:128]: could this be common code in the backend?

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_Mount/AP_Mount_SoloGimbal.cpp [hunk 3]
```diff
@@ -115,15 +115,31 @@ void AP_Mount_SoloGimbal::set_mode(enum MAV_MOUNT_MODE mode)
     _mode = mode;
 }
 
-// send_mount_status - called to allow mounts to send their status to GCS using the MOUNT_STATUS message
-void AP_Mount_SoloGimbal::send_mount_status(mavlink_channel_t chan)
+// send a GIMBAL_DEVICE_ATTITUDE_STATUS message to GCS
+void AP_Mount_SoloGimbal::send_gimbal_device_attitude_status(mavlink_channel_t chan)
```
Comment (peterbarker) [inline libraries/AP_Mount/AP_Mount_SoloGimbal.cpp:119]: Make this non-virtual and create `virtual bool AP_Mount_Backend::get_attitude_quaterniion(Quatnerion &quat)`

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 4]
```diff
@@ -944,7 +944,6 @@ ap_message GCS_MAVLINK::mavlink_id_to_ap_message_id(const uint32_t mavlink_id) c
         { MAVLINK_MSG_ID_BATTERY2,              MSG_BATTERY2},
         { MAVLINK_MSG_ID_CAMERA_FEEDBACK,       MSG_CAMERA_FEEDBACK},
 #if HAL_MOUNT_ENABLED
-        { MAVLINK_MSG_ID_MOUNT_STATUS,          MSG_MOUNT_STATUS},
```
Comment (peterbarker) [inline libraries/GCS_MAVLink/GCS_Common.cpp:947]: Where is the replacement entry here for the replacement message?

**Review by peterbarker — DISMISSED** [review]

File: libraries/GCS_MAVLink/GCS_Common.cpp (hunk 4 above)
Comment (rmackay9) [reply libraries/GCS_MAVLink/GCS_Common.cpp:947]: txs, added the mapping between MAVLINK_MSG_ID_DEVICE_ATTITUDE_STATUS and MSG_DEVICE_ATTITUDE_STATUS.

File: libraries/AP_Mount/AP_Mount_SoloGimbal.cpp (hunk 3 above)
Comment (rmackay9) [reply libraries/AP_Mount/AP_Mount_SoloGimbal.cpp]: done, txs

File: libraries/AP_Mount/AP_Mount_SoloGimbal.cpp (hunk 2 above)
Comment (rmackay9) [reply libraries/AP_Mount/AP_Mount_SoloGimbal.cpp]: yes, txs.  moved to common code in the backend.

**Review by tridge — APPROVED** [review]

**Review by peterbarker — APPROVED** [review]

**Pull request merged**